charmhub/client.go

```diff
@@ -313,3 +222,10 @@ func (c *Client) DownloadResource(ctx context.Context, resourceURL *url.URL) (r
 func (c *Client) ListResourceRevisions(ctx context.Context, charm, resource string) ([]transport.ResourceRevision, error) {
 	return c.resourcesClient.ListResourceRevisions(ctx, charm, resource)
 }
+
+type noopLogger struct{}
+
+func (noopLogger) IsTraceEnabled() bool { return false }
+
+func (noopLogger) Errorf(string, ...interface{}) {}
+func (noopLogger) Tracef(string, ...interface{}) {}
```

Comment:
We need to require a logger rather than add a noop one.

Reply:
Done.